### PR TITLE
Caching of articles for configurations

### PR DESCRIPTION
**Pull request by Feikoritsema**
**Warning, this requires the updated API and database.!** 

To update the database, don't run the entire tools/sql/search_db_migration.sql, but just the last CREATE TABLE to create the caching table. 

This increases speed a lot:)

### CODE REVIEW

File: zeeguu/content_recommender/mixed_recommender.py [hunk 1]
```diff
@@ -33,31 +32,25 @@ def user_article_info(user: User, article: Article, with_content=False):
     return ua_info
 
 
-def article_recommendations_for_user(user, count):
+def add_hash(user, session):
```
Comment (mircealungu) [inline zeeguu/content_recommender/mixed_recommender.py]: is it right to say that this method `recomputes_recommender_cache_if_needed`?

File: zeeguu/content_recommender/mixed_recommender.py [hunk 2]
```diff
@@ -213,3 +238,34 @@ def get_articles_for_search_term(search_term):
...
+    user_source_subscriptions = RSSFeedRegistration.feeds_for_user(user)
+    sources = [rss_feed_id.rss_feed for rss_feed_id in user_source_subscriptions]
+    languages = UserLanguage.all_reading_for_user(user)
+    user_search_filters = SearchFilter.all_for_user(user)
+    search_filters = [search_id.search for search_id in user_search_filters]
+    user_searches = SearchSubscription.all_for_user(user)
+    searches = [search_id.search for search_id in user_searches]
+
+    # This is done because the articles are retrieved for sources if there are any sources,
+    # the languages don't have any effect when there are sources selected.
```
Comment (mircealungu) [inline zeeguu/content_recommender/mixed_recommender.py]: languages have no effect when sources are selected. 
this feels like something that could be a small Hint in the UI :)

File: zeeguu/content_recommender/mixed_recommender.py [hunk 3]
```diff
@@ -213,3 +238,34 @@ def get_articles_for_search_term(search_term):
         return [article for article in search_articles_first if article in search_articles_second]
 
     return ArticleWord.get_articles_for_word(search_terms[0])
+
+
+def get_articles_hash(user):
```
Comment (mircealungu) [inline zeeguu/content_recommender/mixed_recommender.py]: this computes the `reading_preferences_hash` of the user, right? 
or maybe `reading_preferences_configuration_hash`?

File: zeeguu/content_recommender/mixed_recommender.py (hunk 2 above)
Comment (Feikoritsema) [reply zeeguu/content_recommender/mixed_recommender.py]: Good point! Will do, and I'll try to add the topics to the articles :)

File: zeeguu/model/article.py [hunk 4]
```diff
@@ -93,7 +93,7 @@ def vote_broken(self):
     def topics_as_string(self):
         topics = ""
         for topic in self.topics:
-            topics += topic.title + ", "
+            topics += topic.title + " "
```
Comment (mircealungu) [inline zeeguu/model/article.py]: good point 👍

File: zeeguu/content_recommender/mixed_recommender.py [hunk 5]
```diff
@@ -33,31 +32,25 @@ def user_article_info(user: User, article: Article, with_content=False):
     return ua_info
 
 
-def article_recommendations_for_user(user, count):
+def recompute_recommender_cash_if_needed(user, session):
```
[the remark this answers is not in the document]
Comment (mircealungu) [reply zeeguu/content_recommender/mixed_recommender.py]: ah, nevermind. i'll do this refactorign myself and then merge.

**Comment by Feikoritsema** [comment]
The WiFi in my home is messing with me, trying to fix it to be able to push. Don’t forget that when the core is merged, the last create table in the sql migration for search has to be executed and the api should also be merged.

Kind regards,

Feiko Ritsema

Op 21 jun. 2018 om 13:41 heeft Mircea Filip Lungu <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


@mircealungu commented on this pull request.

________________________________

In zeeguu/content_recommender/mixed_recommender.py<https://github.com/zeeguu-ecosystem/Zeeguu-Core/pull/85#discussion_r197102123>:

> @@ -33,31 +32,25 @@ def user_article_info(user: User, article: Article, with_content=False):
     return ua_info


-def article_recommendations_for_user(user, count):
+def recompute_recommender_cash_if_needed(user, session):


ah, nevermind. i'll do this refactorign myself and then merge.

**Comment by mircealungu** [comment]
@Feikoritsema  - please make sure to pull the latest version. i've made the same rename as you did, and we had a conflict. i've fixed it, so it should be ok.

**Pull request merged**